packages/netlify/src/index.ts

```diff
@@ -1,2 +1,182 @@
-export { netlifyFunctions as default, netlifyFunctions } from './integration-functions.js';
-export { netlifyStatic } from './integration-static.js';
+import type { AstroConfig, AstroIntegration, RouteData } from 'astro';
+import { writeFile, rmdir, mkdir, appendFile } from 'fs/promises';
+import { fileURLToPath } from 'url';
+import { build } from 'esbuild';
+import { createRedirectsFromAstroRoutes } from '@astrojs/underscore-redirects';
+import { version as packageVersion } from '../package.json';
+import type { Context } from '@netlify/functions';
+
+export interface NetlifyLocals {
+	netlify: {
+		context: Context;
+	};
+}
+
+const isStaticRedirect = (route: RouteData) => route.type === 'redirect' && route.redirectRoute;
+
+export default function netlifyIntegration(): AstroIntegration {
+	let _config: AstroConfig;
+	let outDir: URL;
+	let rootDir: URL;
+	let astroMiddlewareEntryPoint: URL | undefined = undefined;
+
+	const ssrOutputDir = () => new URL('./.netlify/functions-internal/ssr/', rootDir);
+	const middlewareOutputDir = () => new URL('.netlify/edge-functions/middleware/', rootDir);
+
+	const cleanFunctions = async () => {
+		try {
+			await rmdir(middlewareOutputDir());
+		} catch {}
+		try {
+			await rmdir(ssrOutputDir());
+		} catch {}
+	};
+
+	async function writeRedirects(routes: RouteData[], dir: URL) {
+		const redirects = createRedirectsFromAstroRoutes({
+			config: _config,
+			dir,
+			routeToDynamicTargetMap: new Map(
+				routes
+					.filter(isStaticRedirect) // all other routes are handled by SSR
+					.map((route) => [route, '/.netlify/functions/ssr/']) // we don't really want any redirect to point to SSR, but we need to provide a target and this is a good fallback
+			),
+		});
+
+		if (!redirects.empty()) {
+			await appendFile(new URL('_redirects', outDir), '\n' + redirects.print() + '\n');
+		}
+	}
+
+	async function writeSSRFunction() {
+		await writeFile(
+			new URL('./ssr.mjs', ssrOutputDir()),
+			`
+				import ssrRoute from './entry.mjs';
+				export default ssrRoute;
+				export const config = { name: "Astro SSR", generator: "@astrojs/netlify@${packageVersion}", path: "/*", preferStatic: true };
```

Comment:
Also can we exclude any paths here?

Reply:
doesn't work for edge functions, but this is a netlify function. because we have `preferStatic`, I don't think we need to exclude anything in here - this will be the ultimate fallback, and render a 404 page if there's no matching page.

Reply:
ahh good catch! updated in https://github.com/Skn0tt/adapters/pull/1/commits/18145ca2189fe4947170e3c48e50249908e8e1e1